src/main/java/frc/robot/subsystems/vision/PhotonSubsystem.java

```diff
@@ -0,0 +1,153 @@
+      new DebugEntry<Double>(0.0, "Distance To Tag 3 (m)", this);
+  private DebugEntry<Double> distanceEntryTag4 =
+      new DebugEntry<Double>(0.0, "Distance To Tag 4 (m)", this);
+
+  public PhotonSubsystem(BiConsumer<Pose2d, Double> measurementConsumer) {
+    this.measurementConsumer = measurementConsumer;
+    camera = new PhotonCamera("Camera_Module_v1");
+    while (!camera.isConnected()) {
+      camera = new PhotonCamera("Camera_Module_v1");
+    }
+    result = camera.getLatestResult();
+    aprilTagFieldLayout = AprilTagFields.k2024Crescendo.loadAprilTagLayoutField();
+    poseEstimator =
+        new PhotonPoseEstimator(
+            aprilTagFieldLayout,
+            PoseStrategy.MULTI_TAG_PNP_ON_COPROCESSOR,
+            camera,
+            new Transform3d( // camera offsets will go here when we have them finalized
+                new Translation3d(0, 0, 0),
+                new Rotation3d(
+                    0,
+                    Constants.TurretConstants.LIMELIGHT_PITCH_RADIANS,
+                    Units.degreesToRadians(180))));
+  }
+
+  private EstimatedRobotPose getPVEstimatedPose() {
+    final Optional<EstimatedRobotPose> x = poseEstimator.update();
+    // if (x.isEmpty()) {
+    //   return null;
+    // }
+    return x.get();
+  }
+
+  public Pose3d getPose3d() {
+    return getPVEstimatedPose().estimatedPose;
+  }
+
+  public Pose2d getPose2d() {
+    Pose3d estimatedPose = getPVEstimatedPose().estimatedPose;
+    return new Pose2d(
+        estimatedPose.getX(),
+        estimatedPose.getY(),
+        new Rotation2d(estimatedPose.getRotation().getZ()));
+  }
+
+  public double getTime() {
+    return getPVEstimatedPose().timestampSeconds;
+  }
+
+  public double getTurretYaw(int id) {
+    result = camera.getLatestResult();
+    if (result.hasTargets()) {
+      targets = result.getTargets();
+      for (PhotonTrackedTarget target : targets) {
+        if (target.getFiducialId() == id) {
+          return target.getYaw();
+        }
+      }
+    }
+    return 0;
+  }
+
+  public double getTurretPitch(int id) {
+    result = camera.getLatestResult();
+    if (result.hasTargets()) {
+      targets = result.getTargets();
+      for (PhotonTrackedTarget target : targets) {
+        if (target.getFiducialId() == id) {
+          return target.getPitch();
+        }
+      }
+    }
+    return 0;
+  }
+
+  public void periodic() {
+    if (Robot.isReal()) {
+      result = camera.getLatestResult();
+      resultLatency = result.getLatencyMillis();
+      if (result.hasTargets()) {
+        targets = result.getTargets();
+        if (targets.size() > 1) {
+          measurementsUsed++;
+          measurementConsumer.accept(getPose2d(), getTime() - (resultLatency / 1000));
+          if (measurementsUsed % 100 == 0) {
+            measurementEntry.log(measurementsUsed);
+          }
+        }
+        // logging stuff
+        for (PhotonTrackedTarget target : targets) {
```

Comment:
I still like these calculations, but wondering if we should put them inside of an isCompetiton block.

THoughts?